core/txpool/legacypool/legacypool.go

```diff
@@ -943,8 +943,11 @@ func (pool *LegacyPool) Add(txs []*types.Transaction, sync bool) []error {
 		news = make([]*types.Transaction, 0, len(txs))
 	)
 	for i, tx := range txs {
+		pool.mu.Lock()
+		hash := pool.all.Get(tx.Hash())
+		pool.mu.Unlock()
```

Comment:
It will murder the performance.

Please implement a Clear function in lookup and call it in legacyPool.Clear

Reply:
Good point. Updated in https://github.com/ethereum/go-ethereum/pull/31641/commits/847532f28710c376c6fa8e1ff4c040a75c62c1b8.